components/p3a/brave_p3a_service.cc

```diff
@@ -99,21 +137,92 @@ BraveP3AService::~BraveP3AService() = default;
 void BraveP3AService::RegisterPrefs(PrefRegistrySimple* registry,
                                     bool first_run) {
   BraveP3ALogStore::RegisterPrefs(registry);
-  registry->RegisterTimePref(kLastRotationTimeStampPref, {});
+  // Using "year ago" as default value to fix macOS test crashes
+  const base::Time year_ago = base::Time::Now() - base::Days(365);
+  registry->RegisterTimePref(kLastTypicalRotationTimeStampPref, year_ago);
+  registry->RegisterTimePref(kLastExpressRotationTimeStampPref, year_ago);
   registry->RegisterBooleanPref(kP3AEnabled, true);
 
   // New users are shown the P3A notice via the welcome page.
   registry->RegisterBooleanPref(kP3ANoticeAcknowledged, first_run);
+
+  registry->RegisterDictionaryPref(kDynamicMetricsDictPref);
 }
 
 void BraveP3AService::InitCallbacks() {
-  for (base::StringPiece histogram_name : p3a::kCollectedHistograms) {
+  for (const base::StringPiece& histogram_name :
+       p3a::kCollectedTypicalHistograms) {
     histogram_sample_callbacks_.push_back(
         std::make_unique<
             base::StatisticsRecorder::ScopedHistogramSampleObserver>(
             std::string(histogram_name),
-            base::BindRepeating(&BraveP3AService::OnHistogramChanged, this)));
+            base::BindRepeating(&BraveP3AService::OnHistogramChanged,
```

Comment:
why `this` is replaced with `base::Unretained(this)`?

Reply:
actually it seems once upstream switched to `ScopedHistogramSampleObserver` we can drop the refcounted thing (if we register *all* callbacks on the UI). In this case, callbacks would only fire on UI, and we own them, so Unretained is safe